common/sphinx.h

```diff
@@ -83,14 +83,24 @@ struct hop_data {
 	struct short_channel_id channel_id;
 	struct amount_msat amt_forward;
 	u32 outgoing_cltv;
-	/* Padding omitted, will be zeroed */
-	u8 hmac[HMAC_SIZE];
+};
+
+enum sphinx_payload_type {
+	SPHINX_V0_PAYLOAD = 0,
+	SPHINX_RAW_PAYLOAD = 255,
```

Comment:
Whenever we overload values for internal usage, I always worry about leakage (someone sends us a realm 255 onion in future).

Let's make `route_step.realm` and int, and use negative values for internal ones maybe?  Or 256?